graph/mongo/quadstore.go

```diff
@@ -113,38 +114,57 @@ func ensureIndexes(db *mgo.Database) error {
 }
 
 func createNewMongoGraph(addr string, options graph.Options) error {
-	conn, err := mgo.Dial(addr)
+	conn, err := dialMongo(addr, options)
 	if err != nil {
 		return err
 	}
 	defer conn.Close()
 	conn.SetSafe(&mgo.Safe{})
+	db := conn.DB("")
+	return ensureIndexes(db)
+}
+
+func dialMongo(addr string, options graph.Options) (*mgo.Session, error) {
+	var dialInfo mgo.DialInfo
+	dialInfo.Addrs = strings.Split(addr, ",")
+	user, ok, err := options.StringKey("username")
+	switch {
```

Comment:
Agree. In some cases in stdlib switches are used in small condition chains, but these generally refer to things that would be considered sets of cases on the switch expressionrather than what we see here. There doesn't even need to be if else, since the failing error case returns (can it return a nil?), then you do the work if ok is true.